added more docs to readme (#21)
vincenzopalazzo/cln4go: docs/MAINTAINERS.md

Each commit message consists of a **header**, a **body** and a **footer**. The header has a special
format that includes a **type**, a **scope** and a **subject**:

```
<type>(<scope>): <subject>
<BLANK LINE>
<body>
<BLANK LINE>
<footer>
```

The **header** is mandatory while the **scope** of the header is optional.

All lines in a commit message should be at most 100 characters! This ensures better readability on GitHub as well as in various git tools.

The footer should contain a [closing reference to an issue](https://help.github.com/articles/closing-issues-via-commit-messages/) if any.

Some couple of examples are:

```
docs(changelog): update changelog to beta.5
```

```
fix(release): need to depend on the latest rxjs and zone.js

The version in our package.json gets copied to the one we publish, and users need the latest of these.
```

### Types

- **feat**: A new feature
- **fix**: A bug fix
- **deprecate**: Deprecate a feature and start to the removing process (3 official release or 1 major release)
- **remove**: End of life for the feature.

### Scopes

- **rpc**: Changes related to the core lightning RPC wrapper library

> - **plugin**: Changes related to the plugin library

Reviewer: ```suggestion
- **plugin**: Changes related to the plugin module
- **common**: Changes related to the common module.
```

the search engine indexes it.

An example of commit body is the following one

```
checker: fixes overloading operation when the type is optimized

The stacktrace is the following one

} expected `Foo` not `Foo` - both operands must be the same type for operator overloading
   11 | }
   12 |
   13 | fn (_ Foo) == (_ Foo) bool {
      |                  ~~~
   14 |     return true
   15 | }---
description: "`Rust core lightning Rust framework` HACKING guide"
---
```

## How to make the release

This is the most fun part and also is the most difficult one in a monorepo repository.

In particular, "core lightning Go Framework" has the possibility to release one single package at a time, or
all together.

To prepare for the release, these steps must be followed:

- Bump the version number in the package before the release, and the version inside the `changelog.json` in the package root;
    - Use a Personal Access Token or generate a [new one](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token).
- Generate the changelog related to the package:
    - `export GITHUB_TOKEN="your_token"`

>     - `make {changelog_rpc|changelog_plugin|changelog_plugin_macros|changelog_common|changelog}`, where`make {changelog_client|changelog_flutter|changelog}`, where

Reviewer: ```suggestion
    - `make {changelog_rpc|changelog_plugin|changelog_common|changelog}`, where`:
```

        - `changelog_client`: generate the changelog for rpc library;

>         - `changelog_plugin`: generate the changelog for plugin library;

Reviewer: ```suggestion
        - `changelog_plugin`: generate the changelog for plugin module;
        - `changelog_plugin_macros`: generate the changelog for common module.
```

    - Use a Personal Access Token or generate a [new one](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token).
- Generate the changelog related to the package:
    - `export GITHUB_TOKEN="your_token"`

>     - - `make {changelog_rpc|changelog_plugin|changelog_common|changelog}`, where`:changelog_common|changelog}`, where`make {changelog_client|changelog_flutter|changelog}`, where

Reviewer: ```suggestion
    - `make {changelog_rpc|changelog_plugin|changelog_common|changelog}`, where:
```
